lib/koans.ex

```diff
@@ -82,11 +82,32 @@ defmodule Koans do
   end
 
   defmacro __before_compile__(env) do
-    koans = Module.get_attribute(env.module, :koans) |> Enum.reverse
+    koans = koans(env)
+    intro = extract_into(env)
     quote do
       def all_koans do
         unquote(koans)
       end
+
+      def intro do
+        unquote(intro)
```

Comment:
Is all this metaprogramming needed? Couldn't his function just be something like

```elixir
def intro, do: @intro
```